common/json.hpp

```diff
@@ -27,23 +27,33 @@ namespace mender::common::json {
 
 using namespace std;
 
+namespace error = mender::common::error;
+
 enum JsonErrorCode {
 	NoError = 0,
 	ParseError,
 	KeyError,
 	IndexError,
 	TypeError,
 };
-using JsonError = mender::common::error::Error<JsonErrorCode>;
 
-using ExpectedString = mender::common::expected::Expected<string, JsonError>;
-using ExpectedInt = mender::common::expected::Expected<int, JsonError>;
-using ExpectedBool = mender::common::expected::Expected<bool, JsonError>;
-using ExpectedSize = mender::common::expected::Expected<size_t, JsonError>;
+class JsonErrorCategoryClass : public std::error_category {
+public:
+       const char* name() const noexcept override;
+       string message(int code) const override;
+};
+extern const JsonErrorCategoryClass JsonErrorCategory;
+
+error::Error MakeError(JsonErrorCode code, const string &msg);
+
+using ExpectedString = mender::common::expected::Expected<string, error::Error>;
+using ExpectedInt = mender::common::expected::Expected<int, error::Error>;
+using ExpectedBool = mender::common::expected::Expected<bool, error::Error>;
+using ExpectedSize = mender::common::expected::Expected<size_t, error::Error>;
```

Comment:
I think these four should actually go to `error.hpp` and probably directly into the `mender::common` namespace.

Reply:
The common ones, yes, that's a good idea.

Reply:
Done! Note that the resulting code actually became longer, because we still have to use `using` to shorten their names where we use them. But it's still nicer I think.